### DISCUSSION TITLE
Bash login shell

### DISCUSSION THREAD

**Discussion by kriswuollett** [Eugeny/russh#334] [Q&A]
I assume this question has more to do with `bash` than with `russh`, but how would one get a `bash` login shell which sources the user's `.bashrc` (Ubuntu 22.04).

I've adapted `client_exec_simple` and running `["shopt"]` gets `login_shell     off`. Using `["bash", "-l", "-c", "shopt"]` gets `login_shell     on`. However if I try to run a command, like `["bash", "-l", "-c", "my-app"]` I get exit status 127. If I check with `["bash", "-l", "-c", "echo $PATH"]` it is clear the path for the command isn't in the list.

If I just use the full path to `my-app` it works fine.

I've also tried running the channel `request_pty` before `exec` but it doesn't seem to make a difference.

**Comment by Eugeny** [comment]
You probably also need `-i` (interactive shell)
